File: src/com/eagle/programmar/Django/Terminals/Django_Comment.java
// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Dec 5, 2015

package com.eagle.programmar.Django.Terminals;

import com.eagle.parsers.EagleFileReader;
import com.eagle.parsers.EagleLineReader;
import com.eagle.tokens.TerminalCommentToken;

public class Django_Comment extends TerminalCommentToken
{
	// Need a default constructor for the parser
	public Django_Comment()
	{
		this("");
	}
	
	public Django_Comment(String comment)
	{
		super(comment);
	}
	
	@Override
	public boolean parse(EagleFileReader lines)
	{
		if (findStart(lines) == FOUND.EOF) return false;
		EagleLineReader rec = lines.get(_currentLine);
		return commentUntilSentinel(lines, rec, "{%");
	}
}

File: csrc/com/eagle/programmar/Django/Terminals/Django_Comment.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Dec 5, 2015

namespace com.eagle.programmar.Django.Terminals
{
	using EagleFileReader = com.eagle.parsers.EagleFileReader;
	using EagleLineReader = com.eagle.parsers.EagleLineReader;
	using TerminalCommentToken = com.eagle.tokens.terminals.TerminalCommentToken;

	public class Django_Comment : TerminalCommentToken
	{
		// Need a default constructor for the parser
		public Django_Comment() : this("")
		{
		}

		public Django_Comment(string comment) : base(comment)
		{
		}

		public override bool parse(EagleFileReader lines)
		{
			if (findStart(lines) == FOUND.EOF)
			{
				return false;
			}
			EagleLineReader rec = lines.get(_currentLine);
			return base.possibleCommentPair2(lines, rec, "{#", "#}");
		}

		public override string description()
		{
			return "{# comment #}";
		}
	}

}
